tests/reconfiguration-tests.bats

```diff
@@ -1,46 +1,46 @@
 #!/usr/bin/env bats
 
-source $BATS_TEST_DIRNAME/common.bash
+setup() {
+	load common.bash
+	wait_pods
+}
 
-setup_file() {
-	run kubectl --context $CLUSTER_CONTEXT delete --wait --timeout $TIMEOUT --ignore-not-found clusteradmissionpolicies --all
-	run kubectl --context $CLUSTER_CONTEXT delete --wait --timeout $TIMEOUT --ignore-not-found -f $RESOURCES_DIR/violate-privileged-pod-policy.yaml
-	run kubectl --context $CLUSTER_CONTEXT delete --wait --timeout $TIMEOUT --ignore-not-found -f $RESOURCES_DIR/not-violate-privileged-pod-policy.yaml
-	run kubectl --context $CLUSTER_CONTEXT wait --timeout $TIMEOUT --for=condition=Ready --namespace $NAMESPACE pods --all
+teardown_file() {
+	kubectl delete --wait --timeout $TIMEOUT --ignore-not-found clusteradmissionpolicies --all
+	kubectl delete --wait --timeout $TIMEOUT --ignore-not-found -f $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+	kubectl delete --wait --timeout $TIMEOUT --ignore-not-found -f $RESOURCES_DIR/not-violate-privileged-pod-policy.yaml
 }
 
 @test "[Reconfiguration tests] Test apply policy" {
 	apply_cluster_admission_policy $RESOURCES_DIR/privileged-pod-policy.yaml
 }
 
 @test "[Reconfiguration tests] Test apply pod which violate a policy" {
-	kubectl_apply_should_fail  $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+	kubectl_apply_should_fail $RESOURCES_DIR/violate-privileged-pod-policy.yaml
 }
 
 @test "[Reconfiguration tests] Test apply pod which does not violate a policy" {
-	kubectl_apply_should_succeed $RESOURCES_DIR/not-violate-privileged-pod-policy.yaml
+	kubectl_apply $RESOURCES_DIR/not-violate-privileged-pod-policy.yaml
 }
 
 @test "[Reconfiguration tests] Reconfigure Kubewarden stack" {
-	run helm upgrade --wait --namespace $NAMESPACE --reuse-values  \
-		--kube-context $CLUSTER_CONTEXT \
+	helm upgrade --wait --namespace $NAMESPACE --reuse-values  \
 		--values=$RESOURCES_DIR/reconfiguration-values.yaml \
 		$KUBEWARDEN_CONTROLLER_CHART_RELEASE $CONTROLLER_CHART
-	[ "$status" -eq 0 ]
```

Comment:
why not check the retcode? It could happen that a helm upgrade failed, so there was no change to the policy-server, which means that the running policies keep being active and unchanged.